pkg/kube/client.go

```diff
@@ -884,6 +887,25 @@ func (c *client) GetIstioVersions(ctx context.Context, namespace string) (*versi
 	return &res, errs
 }
 
+func (c *client) GetProxyPods(ctx context.Context, limit int64, token string) (*v1.PodList, error) {
+	opts := metav1.ListOptions{
+		LabelSelector: label.ServiceCanonicalName.Name,
+		FieldSelector: "status.phase=Running",
+		Limit:         limit,
+		Continue:      token,
+	}
+	if c.revision != "" {
+		opts.LabelSelector = fmt.Sprintf("%s,%s=%s", label.ServiceCanonicalName.Name, label.IoIstioRev.Name, c.revision)
```

Comment:
%s,%s=%s is this right?

Reply:
Yes. 

Please look at https://github.com/istio/istio/blob/abfd63329f7fbdaaea2b0a9e5d66ec86bb87947f/pkg/kube/client.go#L819.
It is for finding the Istiod pod, but the label is the same.

Reply:
Hopefully that isn't used anywhere important since its not correct. A sidecar pod doesn't necessarily have the istio.io/rev label

Reply:
@howardjohn I have been confused. Thanks for pointing out.

I changed the logic to use the annotated injection status to identify "revision".
(It uses pretty dumb loop, though)

If we have a label for distinguishing the revision at the injected pod, it seems to be handy.